solutions/rust/07-expiry/code/src/store.rs

```diff
@@ -0,0 +1,49 @@
+use std::collections::HashMap;
+use std::time::{Duration, Instant};
+
+pub struct Store {
+    data: HashMap<String, Entry>,
+}
+
+struct Entry {
+    t: Option<Instant>,
+    value: String,
+}
+
+impl Store {
+    pub fn new() -> Self {
+        Store {
+            data: HashMap::new(),
+        }
+    }
+
+    pub fn set(&mut self, key: String, value: String) {
+        let entry = Entry { t: None, value };
+        self.data.insert(key, entry);
+    }
+
+    pub fn set_px(&mut self, key: String, value: String, px: u64) {
```

Comment:
@ThetaSinner how about naming this `set_with_expiry`, and naming `px` -> `expiry_ms` / `expiry_in_milliseconds`? Seems like the Redis command terminology is leaking into a layer where it isn't really relevant - thoughts?